lib/ex_doc/formatter/markdown/templates.ex

```diff
@@ -0,0 +1,244 @@
+defmodule ExDoc.Formatter.MARKDOWN.Templates do
+  @moduledoc false
+
+  require EEx
+
+  import ExDoc.Utils,
+    only: [before_closing_body_tag: 2, h: 1, text_to_id: 1]
+
+  @doc """
+  Generate content from the module template for a given `node`.
+  """
+  def module_page(config, module_node) do
+    summary =
+      for group <- module_node.docs_groups do
+        {group.title, group.docs}
+      end
+
+    module_template(config, module_node, summary)
+  end
+
+  @doc """
+  Returns the formatted title for the module page.
+  """
+  def module_type(%{type: :task} = _node), do: ""
+  def module_type(%{type: :module} = _node), do: ""
+  def module_type(%{type: type} = _node), do: "(#{type})"
+
+  @doc """
+  Format the attribute type used to define the spec of the given `node`.
+  """
+  def format_spec_attribute(module, node) do
+    module.language.format_spec_attribute(node)
+  end
+
+  @doc """
+  Generated ID for static file.
+  """
+  def static_file_to_id(static_file) do
+    static_file |> Path.basename() |> text_to_id()
+  end
+
+  def node_doc(%{doc: doc}) when is_list(doc) do
+    # Handle DocAST by converting to markdown
+    ExDoc.DocAST.to_markdown(doc)
+  end
+
+  def node_doc(%{doc: doc}) when is_binary(doc), do: doc
+  def node_doc(%{source_doc: %{"en" => source}}) when is_binary(source), do: source
+  def node_doc(%{rendered_doc: source}) when is_binary(source), do: source
+
+  def node_doc(%{source_doc: %{"en" => source}}) when is_list(source) do
+    # Handle DocAST by converting to markdown
+    # For Erlang docs, we can extract text content
+    extract_text_from_doc_ast(source)
+  end
+
+  def node_doc(_), do: nil
+
+  @doc """
+  Get synopsis for a node, handling both DocAST and string documentation.
+  Uses DocAST synopsis extraction logic for consistency with HTML formatter.
+  """
+  def node_synopsis(%{doc: doc}) when is_list(doc) do
+    # For DocAST, extract synopsis DocAST and convert to markdown
+    case extract_synopsis_ast(doc) do
+      nil -> nil
+      synopsis_ast -> ExDoc.DocAST.to_markdown(synopsis_ast)
+    end
+  end
+
+  def node_synopsis(%{doc: doc}) when is_binary(doc) do
+    synopsis(doc)
+  end
+
+  def node_synopsis(%{source_doc: %{"en" => source}}) when is_binary(source) do
+    synopsis(source)
+  end
+
+  def node_synopsis(%{rendered_doc: source}) when is_binary(source) do
+    synopsis(source)
+  end
+
+  def node_synopsis(%{source_doc: %{"en" => source}}) when is_list(source) do
+    # For Erlang DocAST, extract synopsis and convert to plain text
```

Comment:
I think the code is self-explanatory here.
If agreed upon, please remove everywhere else.

```suggestion
```